### PR TITLE
Infra logging using logrus

### PR DESCRIPTION
**Pull request by JaafarRammal**
# Summary

Logging support. Uses https://github.com/Sirupsen/logrus

## Additional Information

- Log to file
- Separate agent, sim, tower logs
- Use fields for easier data processing for people doing ML + for frontend
- Can use JSON and more custom fields in the future

## Test Plan

- N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,8 @@
 
 # Backend build
 bin/
-**/.DS_Store
\ No newline at end of file
+**/.DS_Store
+
+# Logs
+logs/
+**.log
```
Comment (jzzheng22) [inline .gitignore:23]:
```suggestion
**.log

```

Comment (jzzheng22) [reply .gitignore:23]: GitHub complaining about no newline at end of file so I guess fix it

File: cmd/backend/main.go [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+			fmt.Println("error opening file: ", err)
+			return
+		}
+		defer f.Close()
+		_, err = io.Copy(simLog, f)
+		if err != nil {
+			fmt.Println("error copying to simulation log: ", err)
+			return
+		}
+	}()
```
Comment (jzzheng22) [inline cmd/backend/main.go]: I kinda hate this... but I wrote it...

File: cmd/backend/main.go [hunk 3]
```diff
@@ -1,15 +1,67 @@
...
+	"io"
+	"os"
+	"time"
+
 	"github.com/SOMAS2021/SOMAS2021/pkg/simulation"
+	log "github.com/sirupsen/logrus"
 )
 
 func main() {
+	// logger setup
```
Comment (jzzheng22) [inline cmd/backend/main.go:14]:
```suggestion
	// logger setup
	// TODO: clean up logger initialisation and closing code
```

File: cmd/backend/main.go (hunk 2 above)
Comment (JaafarRammal) [reply cmd/backend/main.go]: Added a few comments, magic it looks nice :o

**Review by jzzheng22 — APPROVED** [review]

**Review by hussain2603 — APPROVED** [review]

**Pull request merged**